broker-plugins/amqp-1-0-protocol/src/main/java/org/apache/qpid/server/protocol/v1_0/AMQPConnection_1_0Impl.java

```diff
@@ -1197,6 +1197,9 @@ private void closeConnection(final Error error)
             default:
                 throw new ServerScopedRuntimeException("Unknown state: " + _connectionState);
         }
+
+        getSender().close();
```

Comment:
Closing the network connection at line 1201 looks wrong to me.  If connection state is OPEN, the counterpart should get a chance to respond with a CLOSE perforrmative after sending CLOSE by broker. I would argue that more correct fix for the issue would be to close the network connection only when state is AWAIT_OPEN as illustrated below

`
--- a/broker-plugins/amqp-1-0-protocol/src/main/java/org/apache/qpid/server/protocol/v1_0/AMQPConnection_1_0Impl.java
+++ b/broker-plugins/amqp-1-0-protocol/src/main/java/org/apache/qpid/server/protocol/v1_0/AMQPConnection_1_0Impl.java
@@ -1179,6 +1179,7 @@ public class AMQPConnection_1_0Impl extends AbstractAMQPConnection<AMQPConnectio
                 sendOpen(0, 0);
                 sendClose(close);
                 _connectionState = ConnectionState.CLOSED;
+                getSender().close();
                 break;
             case OPENED:
                 sendClose(close);
`